airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/message/DestinationMessage.kt

```diff
@@ -200,7 +205,62 @@ data class DestinationRecordRaw(
     }
 
     fun asEnrichedDestinationRecordAirbyteValue(): EnrichedDestinationRecordAirbyteValue {
-        TODO()
+        val rawJson = asRawJson()
+
+        // Get the set of field names defined in the schema
+        val schemaFields =
+            when (schema) {
+                is ObjectType -> schema.properties.keys
+                else -> emptySet()
+            }
+
+        val declaredFields = mutableMapOf<String, EnrichedAirbyteValue>()
+        val airbyteMetaFields = mutableMapOf<String, EnrichedAirbyteValue>()
+        val undeclaredFields = mutableMapOf<String, JsonNode>()
+
+        // Process fields from the raw JSON
+        rawJson.fields().forEach { (fieldName, fieldValue) ->
+            when {
+                schemaFields.contains(fieldName) -> {
+                    // Declared field (exists in schema)
+                    val fieldType =
+                        (schema as ObjectType).properties[fieldName]?.type
+                            ?: throw IllegalStateException(
+                                "Field '$fieldName' exists in schema keys but not in properties"
```

Comment:
checking: this should never happen? b/c `schemaFields = (schema as ObjectType).properties.keys`, and we just did a `schemaFields.contains` check?

(maybe better to just do `properties[fieldName]!!`)